src/plugins/intel_gpu/src/graph/impls/ocl_v2/rope_opt.cl

```diff
@@ -331,8 +338,16 @@ KERNEL(rope_opt)
     #endif
     cos_sin_p = gather[gather_idx];
 #endif
+
+#if INPUT1_DIMS == 4 && INPUT2_DIMS == 4
     cos_sin_p = cos_sin_p < INPUT1_SIZE_Y ? cos_sin_p : 0;
+#elif INPUT1_DIMS == 2 && INPUT2_DIMS == 2
+    cos_sin_p = cos_sin_p < INPUT1_BATCH_NUM ? cos_sin_p : 0;
+#else
+#   error "rope_opt.cl - 4 or 2 of INPUT1_DIMS/INPUT2_DIMS is supported only"
```

Comment:
nit: we can keep only one check at the very beginning and avoid its duplication

Reply:
Updated! Thanks!